### ISSUE TITLE
spacemacs/helm-find-files (SPC f f) not correctly autoloading

### ISSUE THREAD

**Issue by jackkamm** [syl20bnr/spacemacs#7325]
#### Description :octocat:

After calling helm-find-files and pressing Ctrl+z to select an action, and typing in some action, the helm buffer is empty and reports an error.

I noticed this when trying to use spacemacs/helm-find-files (SPC-f-f) to grep in my home-directory on startup.

I found that, if I call helm-mini (SPC b b) to switch buffers, this will autoload some stuff that makes
the helm Actions work correctly again.
#### Reproduction guide :beetle:
- Start Emacs
- spacemacs/helm-find-files (SPC f f)
  - (EDIT to add: needs to be called thru SPC f f. calling thru M-x doesn't reproduce the error)
- Control+SPC to select the first item (~/.)
- Control+z to open the action menu
- Start typing "grep"

_Observed behaviour:_ :eyes: :broken_heart:
The list of available actions is empty; the minibuffer reports,
"Helm issued errors: helm-match-from-candidates in source `Actions': wrong-type-argument (stringp nil) [3 times]"

_Expected behaviour:_ :heart: :smile:
A list of available actions matching "grep" should be listed in the helm buffer.
#### System Info :computer:
- OS: darwin
- Emacs: 25.1.1
- Spacemacs: 0.200.0
- Spacemacs branch: develop (rev. ff95f2a)
- Graphic display: t
- Distribution: spacemacs
- Editing style: hybrid
- Completion: helm
- Layers:

``` elisp
(helm my-spacemacs-layer auto-completion emacs-lisp git github org python ipython-notebook ess shell latex bibtex syntax-checking shell-scripts spell-checking version-control)
```

**Comment by mahinshaw** [comment]
@jackkamm I cannot reproduce this.  Is it still an issue?

**Comment by jackkamm** [comment]
@mahinshaw Yes it is still an issue.

I just did fresh install of spacemacs on OSX and Archlinux and reproduced the error on both.

Now that I think of it, I only noticed this after upgrading to emacs 25. What version emacs are you using?

**Comment by mahinshaw** [comment]
I'm on 25.1.1 on OSX.  If you just re cloned your .emacs.d, I'm guessing you have an up to date helm package.  Mysterious

**Comment by jackkamm** [comment]
After some digging around, I found that setting `dotspacemacs-helm-use-fuzzy 'source` fixes this for me.

More specifically, I tracked the error down to `helm-fuzzy-match` in `helm.el` (in helm-core-20170402.607, line 3060-3069). Initially `helm--fuzzy-regexp-cache` is empty, and so `let ((regexp (cadr (gethash 'helm-pattern helm--fuzzy-regexp-cache))))` sets `regexp` to `nil`. Then, the call `(string-match regexp candidate)` returns an error because `string-match` can't handle nil value.

By setting `dotspacemacs-helm-use-fuzzy 'source`, I think this disables the fuzzy matching within the helm-find-files-action-menu, so `helm-fuzzy-match` isn't called and the error is avoided.

I'm a bit surprised no one else has found this issue, it appears for me on both OSX and Arch on a clean spacemacs install. The key to reproducing is to immediately enter the helm-find-files-action-menu before doing anything else in emacs. As soon as `helm-M-x` or `helm-mini` is called, the hashtable `helm--fuzzy-regexp-cache` is filled and `(cadr (gethash 'helm-pattern helm--fuzzy-regexp-cache))` returns the empty string instead of `nil`, averting the error.
